flake.nix

```diff
@@ -0,0 +1,48 @@
+{
+  inputs = {
+    holonix = {
+      url = "github:holochain/holonix?ref=main-0.4";
+      inputs.crane.follows = "crane";
+      inputs.rust-overlay.follows = "rust-overlay";
+    };
+
+    nixpkgs.follows = "holonix/nixpkgs";
+
+    crane.url = "github:ipetkov/crane";
+
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "holonix/nixpkgs";
+    };
+  };
+
+  outputs = inputs@{ nixpkgs, holonix, crane, rust-overlay, ... }:
+    holonix.inputs.flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = builtins.attrNames holonix.devShells;
+
+      perSystem = { inputs', config, system, pkgs, lib, ... }:
+        {
+          formatter = pkgs.nixpkgs-fmt;
+
+          devShells.default = pkgs.mkShell {
+            packages = [
+              (with inputs'.holonix.packages; [
+                holochain
+                lair-keystore
+                rust
```

Comment:
I see, this is going to conflict with a `rust-toolchain.toml` then. I'm not that keen on mixing these. If we're using Nix locally then we should be using it on CI too. I don't think we actually need this though. What we need is sweettest and I think we can entirely do without a flake

Reply:
I see, wasn't sure as well what the nix flake would be doing if we are not exporting a nix package now